pkg/stanza/fileconsumer/matcher/internal/filter/exclude_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package filter
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestExcludeOlderThanFilter(t *testing.T) {
+	twoHoursAgo := time.Now().Add(-2 * time.Hour)
+	threeHoursAgo := twoHoursAgo.Add(-1 * time.Hour)
+
+	cases := map[string]struct {
+		files            []string
+		fileMTimes       []time.Time
+		excludeOlderThan time.Duration
+
+		expect      []string
+		expectedErr string
+	}{
+		"no_files": {
+			files:            []string{},
+			fileMTimes:       []time.Time{},
+			excludeOlderThan: 2 * time.Hour,
+
+			expect:      []string{},
+			expectedErr: "",
+		},
+		"exclude_no_files": {
+			files:            []string{"a.log", "b.log"},
+			fileMTimes:       []time.Time{twoHoursAgo, twoHoursAgo},
+			excludeOlderThan: 3 * time.Hour,
+
+			expect:      []string{"a.log", "b.log"},
+			expectedErr: "",
+		},
+		"exclude_some_files": {
+			files:            []string{"a.log", "b.log"},
+			fileMTimes:       []time.Time{twoHoursAgo, threeHoursAgo},
+			excludeOlderThan: 3 * time.Hour,
```

Comment:
This can be a followup PR but I think the test is set up a bit strange here because we're comparing "exactly 3 hours" to "3 hour + trivial time" (because we initialize `threeHoursAgo` above and compare it just a few cycles later.) I think we should instead make this an unambiguous comparison and consider ways to test the exactly equal case.